Marlin/src/HAL/HAL_STM32F1/HAL_sdio_STM32F1.cpp

```diff
@@ -0,0 +1,279 @@
+/**
+ * Marlin 3D Printer Firmware
+ * Copyright (C) 2016 MarlinFirmware [https://github.com/MarlinFirmware/Marlin]
+ *
+ * Based on Sprinter and grbl.
+ * Copyright (C) 2011 Camiel Gubbels / Erik van der Zalm
+ * Copyright (C) 2017 Victor Perez
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifdef __STM32F1__
+
+#include "HAL_sdio_STM32F1.h"
+
+SDIO_CardInfoTypeDef SdCard;
+
+bool SDIO_Init(void) {
+  uint32_t count = 0U;
+  SdCard.CardType = SdCard.CardVersion = SdCard.Class = SdCard.RelCardAdd = SdCard.BlockNbr = SdCard.BlockSize = SdCard.LogBlockNbr = SdCard.LogBlockSize = 0;
+
+  sdio_begin();
+  sdio_set_dbus_width(SDIO_CLKCR_WIDBUS_1BIT);
+
+  dma_init(SDIO_DMA_DEV);
+  dma_disable(SDIO_DMA_DEV, SDIO_DMA_CHANNEL);
+  dma_set_priority(SDIO_DMA_DEV, SDIO_DMA_CHANNEL, DMA_PRIORITY_VERY_HIGH);
+
+
+  if (!SDIO_CmdGoIdleState()) { return false; }
```

Comment:
If the intention is to ignore the result of the first, doesn't the return false here defeat that? (sorry if I misunderstand what is supposed to happen here).

Reply:
CMD0 does not wait for SD card's response. 
False here means that SDIO hardware failed to send command, which should never be the case when SDIO is properly configured and clocked.